hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/ViewGeneratorState.java

```diff
@@ -91,5 +87,45 @@ public StructuredTrace getTrace() {
     public Map<ByteBuffer, ByteBuffer> getChildToParentEventIds() {
       return childToParentEventIds;
     }
+
+    /**
+     * Note:
+     * As of now, open tracing specification represents two construct for providing support
+     * for relation between spans: child_of and follow_from.
+     * Ref : https://opentracing.io/specification/
+     *
+     * On the other hand, as of now, open telemetry supports only concept of parent
+     * Ref : https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto#L85
+     *
+     * However, there are open discussion for similar construct on otel community.
+     * Ref: https://github.com/open-telemetry/opentelemetry-specification/issues/65
+     *
+     * As, both the construct `child_of` and `follow_from` represent parent-child relation in common
+     * where in one case parent is interested in child span's result while in other case not.
+     *
+     * So, to support common behaviour, we will be establish link for `follow_from` as well.
+     *
+     * Also expecting only 1 child-parent relation
+     * */
+
+    @Nullable
+    private ByteBuffer buildParentChildLinks(Event event) {
+      Map<Boolean, List<EventRef>> referenceSplits =
+              event.getEventRefList().stream()
+                      .collect(
+                              Collectors.partitioningBy(
+                                      eventRef -> eventRef.getRefType() == EventRefType.CHILD_OF));
+
+      List<EventRef> childEventRef = referenceSplits.get(true);
+      if (childEventRef != null && childEventRef.size() > 0) {
+        return childEventRef.get(0).getEventId();
+      } else {
+        List<EventRef> followFromEventRef = referenceSplits.get(false);
+        if (followFromEventRef != null && followFromEventRef.size() == 1) {
```

Comment:
can we put the condition `followFromEventRef.size() > 0`?

Reply:
have made it similar for both CHILD_OF and FOLLOW_FROM  > 0